### PR TITLE
NEW - Add permissions verification to access Katka projects

### PR DESCRIPTION
**Pull request by CCardosoDev**

### CODE REVIEW

File: bitbucket/mixins.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        validated_data = serializer.validated_data
+
+        filter = {}
+        for api_field in self.lookup_fields:
+            # translate api_filed name to model field name (if it's not the same)
+            model_field_name = api_field if api_field not in API_TO_MODEL_MAP else API_TO_MODEL_MAP[api_field]
+            filter[model_field_name] = validated_data.get(api_field)
+
+        with bitbucket_exception_to_api():
+            obj = get_object_or_404(self.queryset, **filter)
```
Comment (rodgomes) [inline bitbucket/mixins.py:37]: How are we going to be sure that the user using this endpoint/mixin, actually has permission to read the given katka project?

Comment (rodgomes) [reply bitbucket/mixins.py:37]: Or is it already taken care by `PermissionRequiredMixin `?

Comment (CCardosoDev) [reply bitbucket/mixins.py:37]: When adding the project we have to also give permissions to the specific group(s). I will leave that part to when we have the story to create/delete/update katka projects.

File: bitbucket/mixins.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+from .exceptions import bitbucket_exception_to_api
+from .models import KatkaProject
+
+
+class BitbucketPermissionRequiredMixin(PermissionRequiredMixin, APIView):
+    permission_required = 'view_katkaproject'
+
+    queryset = KatkaProject
+    lookup_fields = ('katka_project_id',)
+    lookup_fields_serializer = serializers.KatkaProjectSerializer
```
Comment (rodgomes) [inline bitbucket/mixins.py:18]: why renaming to `KatkaProjectSerializer`?

File: bitbucket/mixins.py (hunk 1 above)
Comment (CCardosoDev) [reply bitbucket/mixins.py:37]: Yes, indeed, that is the case, it verifies the permissions for that specific object

File: bitbucket/mixins.py (hunk 2 above)
Comment (CCardosoDev) [reply bitbucket/mixins.py:18]: When importing serializer.KatkaProject and models.KatkaProject they have the same, thought about changing the name of it to include Serializer, but I could also make a more specific name for the import.

File: bitbucket/mixins.py (hunk 1 above)
Comment (rodgomes) [reply bitbucket/mixins.py:37]: cool! And that group will be mapped to the ldap group right?

Comment (CCardosoDev) [reply bitbucket/mixins.py:37]: yap, they are exactly the same! Created together with the users (if it does not exist yet)

**Review by rodgomes — APPROVED** [review]

**Pull request merged**